src/server/scripts/BrokenIsles/EyeOfAzshara/boss_king_deepbeard.cpp

```diff
@@ -15,10 +15,214 @@
+            {
+                me->CastSpell(me, SPELL_GASEOUS_BUBBLES, false);
+                events.Repeat(33s);
+                break;
+            }
+            case SPELL_GROUND_SLAM:
+            {
+                if (Unit* target = SelectTarget(SELECT_TARGET_TOPAGGRO))
+                {
+                    me->CastSpell(target, SPELL_GROUND_SLAM, false);
+                    me->CastSpell(me, SPELL_GROUND_SLAM_DAMAGE, true);
+                    events.Repeat(20s);
+                }
+                
+                break;
+            }
+        }
+    }
+
+private:
+    bool _firstQuake = true;
+};
+
+// 193018
+class aura_king_deepbeard_gaseous_bubbles : public AuraScript
+{
+    PrepareAuraScript(aura_king_deepbeard_gaseous_bubbles);
+
+    void OnRemove(AuraEffect const* /*auraEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        if (GetTargetApplication()->GetRemoveMode() == AURA_REMOVE_BY_EXPIRE)
+        {
+            Unit* target = GetTarget();
+            Unit* caster = GetCaster();
+
+            if (caster)
+            {
+                Difficulty instanceDifficulty = caster->GetInstanceScript()->instance->GetDifficultyID();
+                if (instanceDifficulty == DIFFICULTY_HEROIC || instanceDifficulty == DIFFICULTY_MYTHIC)
+                {
+                    int32 damage = GetAura()->GetEffect(EFFECT_0)->GetAmount();
+                    target->CastCustomSpell(target, SPELL_GASEOUS_EXPLOSION, &damage, nullptr, nullptr, true);
+                }
+                else
+                    target->CastSpell(target, SPELL_GASEOUS_EXPLOSION, true);
+            }
+        }
+    }
+
+    void Register() override
+    {
+        OnEffectRemove += AuraEffectRemoveFn(aura_king_deepbeard_gaseous_bubbles::OnRemove, EFFECT_0, SPELL_AURA_SCHOOL_ABSORB, AURA_EFFECT_HANDLE_REAL);
+    }
+};
+
+// 97916
+struct npc_king_deepbeard_quake : public ScriptedAI
+{
+    npc_king_deepbeard_quake(Creature* creature) : ScriptedAI(creature) { }
+
+    void Reset() override
+    {
+        me->setFaction(16);
+        me->CastSpell(me, SPELL_AFTERSHOCK, false);
+    }
+};
+
+// Spell: 193054
+// AT: 9683
+struct at_king_deepbeard_call_the_seas : AreaTriggerAI
+{
+    at_king_deepbeard_call_the_seas(AreaTrigger* areatrigger) : AreaTriggerAI(areatrigger) { }
+
+    void OnUnitEnter(Unit* unit) override
+    {
+        Unit* caster = at->GetCaster();
+        Player* player = unit->ToPlayer();
+
+        if (caster && player && caster->IsValidAttackTarget(player))
+            caster->CastSpell(player, SPELL_CALL_THE_SEAS_DAMAGE, true);
+    }
+
+    void OnUnitExit(Unit* unit) override
+    {
+        unit->RemoveAurasDueToSpell(SPELL_CALL_THE_SEAS_DAMAGE);
+    }
+};
+
+// 193245
+class aura_king_deepbeard_gain_energy : public AuraScript
+{
+    PrepareAuraScript(aura_king_deepbeard_gain_energy);
+
+    void OnPeriodic(AuraEffect const* /*auraEff*/)
+    {
+        if (Unit* caster = GetCaster())
```

Comment:
What does the spell energize by default? Afaik it was energy when you showed me spellwork output

Reply:
Yes it's energy